### ISSUE TITLE
[wg/webperf] Web Performance Working Group Charter

### ISSUE THREAD

**Issue by caribouW3** [w3c/strategy#531]
New charter proposal, reviewers please take note.

# Charter Review

[Charter](https://w3c.github.io/charter-drafts//2026/webperf-wg-charter.html)

[diff from charter template](https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://w3c.github.io/charter-drafts//2026/webperf-wg-charter.html)

Expected end of charter refinement phase: [2026-04-15](https://lists.w3.org/Archives/Member/w3c-ac-members/2026JanMar/0019.html)

If applicable:

[diff from previous charter](https://services.w3.org/htmldiff?doc1=https://www.w3.org/2023/11/webperf-charter-2023.html&doc2=https://w3c.github.io/charter-drafts//2026/webperf-wg-charter.html)

[chair dashboard](https://www.w3.org/PM/Groups/chairboard.html?gid=wg/webperf)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* New
 - [ ] New WG 
 - [ ] New IG
 
* Existing 
 - [x ] Existing WG recharter 
 - [ ] Existing IG recharter
 
Horizontal Reviews: apply the Github label "Horizontal review requested" to request reviews for accessibility (a11y), internationalization (i18n), privacy, security, and TAG. **Also add a "card" for this issue to the [Strategy Funnel](https://github.com/orgs/w3c/projects/97/views/2).**

## Known or potential areas of concern

Where would charter proponents like to see issues raised? (this strategy funnel issue, a different github repo, email, ...)

issues in https://github.com/w3c/charter-drafts/issues

## Anything else we should think about as we review? 

Note: proposed chairs should be copied @... on this issue.
@yoavweiss @nicjansma

Note: The Technical Strategy Team Lead or Project & Process Team Lead will assign the issue to a [Charter Facilitator](https://www.w3.org/guide/process/charter.html#cr-acceptance) for new charters. For rechartering, the team contact is the Charter Facilitator by default, please assign the issue to them directly.

**Comment by tidoust** [comment]
@caribouW3, ok to [initiate charter refinement](https://www.w3.org/guide/process/adv-notice.html).

**Comment by caribouW3** [comment]
Charter updated with a list of deliverable with a lot of @@ To Be Discussed marks.

**Comment by xueyuanjia** [comment]
> [...] ok to [initiate charter refinement](https://www.w3.org/guide/process/adv-notice.html).

Charter refinement started on [2026-01-21](https://lists.w3.org/Archives/Public/public-web-perf/2026Jan/0008.html)

**Comment by ruoxiran** [comment]
no comment or request from APA.

**Comment by himorin** [comment]
No comment or request from i18n

**Comment by dontcallmedom** [comment]
* some of the deliverables document FPWD or WD in "expected completion" which seems odd
* rather than having a "Past specifications" section with the same level of detail as specs on which the group actually intends to work, I would suggest documenting these transitions to WICG/WHATWG in a simple paragraph at the end of the "normative spec" sections
* I would suggest retitling the "Standards" section as "Maintenance" for clarity; with that said, since the said specs already appear in the unleveled versions in the "normative spec" sections, I'm not sure the section itself is even needed
* "other deliverables" mentions "two non-normative documents", but only lists one ("Performance APIs, Security and Privacy document"); that document is then listed again in other non-normative deliverables that may be created
  * s/principals/principles/ presumably in the description of work expected on that doc
  * no work seems to have happened on that document in the past charter - does the group really intend to work on it?
* in coordination, the Privacy IG is now the Privacy WG

**Comment by caribouW3** [comment]
> * some of the deliverables document FPWD or WD in "expected completion" which seems odd

Those are adopted drafts, one is not yet published (Element Timing) and the second one just published 
as FPWD today. So the WG has not worked on timelines yet.

> * rather than having a "Past specifications" section with the same level of detail as specs on which the group actually intends to work, I would suggest documenting these transitions to WICG/WHATWG in a simple paragraph at the end of the "normative spec" sections

The goal was readability of a diff-marked vs. the previous charter, but it seems indeed somewhat 
confusing.
See PR https://github.com/w3c/charter-drafts/pull/796

> * I would suggest retitling the "Standards" section as "Maintenance" for clarity; with that said, since the said specs already appear in the unleveled versions in the "normative spec" sections, I'm not sure the section itself is even needed

Done, see PR 

> * "other deliverables" mentions "two non-normative documents", but only lists one ("Performance APIs, Security and Privacy document"); that document is then listed again in other non-normative deliverables that may be created
>   
>   * s/principals/principles/ presumably in the description of work expected on that doc

both fixed

>   * no work seems to have happened on that document in the past charter - does the group really intend to work on it?
> * in coordination, the Privacy IG is now the Privacy WG

No spec has progressed to "living standard" (CR) state so there was no update to this document, but the goal to keep it in the charter is to update it when new specs reach maturity.

**Comment by plehegar** [comment]
Current draft says:
 "The meetings themselves are open to public and can be attended by any observer, at the Chairs' discretion."

The process defines what a participant is, but not what an observer is. In fact, we introduced the concept of observers for the purpose of TPAC originally (see [How should Working Groups handle contributions from non-participants (e.g., meeting guests or on public lists)?](https://www.w3.org/2003/12/22-pp-faq#non-participants)).

However, the Process also says: "A [Chair](https://www.w3.org/policies/process/#GeneralChairs) may invite an individual with a particular expertise to attend a meeting on an exceptional basis." [3.1.2. Meetings](https://www.w3.org/policies/process/#GeneralMeetings). That's clearly not the case here.

So, I don't believe that the draft charter is compatible with the Process as it is.

In addition, there are other variations from the charter template that are problematic because it's changing terms defined in the W3C Process: s/Member-only/Member-confidential/ , s/non-participants/non-Members/, etc. This charter ought not to do that unless it wants to provide the definitions for all of those terms...

**Comment by caribouW3** [comment]
See https://github.com/w3c/charter-drafts/pull/797 for wording adjustments to align with process prose.

**Comment by plehegar** [comment]
> See [w3c/charter-drafts#797](https://github.com/w3c/charter-drafts/pull/797) for wording adjustments to align with process prose.

More changes than that. This charter needs to be better aligned with the charter template.

**Comment by caribouW3** [comment]
[TAG feedback](https://github.com/w3ctag/design-reviews/issues/1196#issuecomment-4351217509)

About the Deliverables section structure:
https://github.com/w3c/charter-drafts/pull/796/  already fixed this partially, answering [Dom's comment above](https://github.com/w3c/strategy/issues/531#issuecomment-4332903983)

About Moving specs to incubation:
It has been subject to discussions inside the WG, at TPAC or during a call. The AC review on the new charter ensures that we don't push aside specs that should not. I don't think we can derive criteria from the past occurrences of dropping work. Rechartering seems to be the best time to get feedback on discontinuing work, although the [W3C Process (section on abandoning unfinished work)](https://www.w3.org/policies/process/#abandon-draft) allows a WG to do that by itself.

About Single engine vs. single browser: 
Indeed it means a single engine. Fixed in 
https://github.com/w3c/charter-drafts/commit/1238ee3f5a3b61e2c5f5e62f031b56482d4b809b

About Features with fewer than 2 implementers:
It is a difficult question, as it happens frequently that some implementers somewhat "abstain" from giving a commitment, without objecting to starting to develop the solution. After discussion in the Working Group, it appeared that the current charter template has an unclear wording to some (it actually applies to specs that already reached CR, but it does not say so). The WG preferred this wording that clarifies that we'll use the at-risk mechanism and mark experimental features when there is uncertainty about implementation plans by at least 2 engines.

**Comment by caribouW3** [comment]
> > See [w3c/charter-drafts#797](https://github.com/w3c/charter-drafts/pull/797) for wording adjustments to align with process prose.
> 
> More changes than that. This charter needs to be better aligned with the charter template.

Please review the updated PR.

**Comment by dontcallmedom** [comment]
A couple of editorial fixes are still needed:

> * "other deliverables" mentions "Performance APIs, Security and Privacy document"; that document is then listed again in other non-normative deliverables that may be created
> * in coordination, the Privacy IG is now the Privacy WG

**Comment by caribouW3** [comment]
Fixed (https://github.com/w3c/charter-drafts/commit/18d0268812f9ba52b7edd0ec8ce02c5d6fb4a6b2)
Thanks!

**Comment by plehegar** [comment]
[Proposed W3C Charter: Web Performance Working Group (until 2026-07-01)](https://lists.w3.org/Archives/Public/public-new-work/2026Jun/0001.html)

**Comment by bluesmoon** [comment]
Should [Container Timing](https://github.com/WICG/container-timing) be included?

**Comment by plehegar** [comment]
Added to [FO Dashboard](https://www.w3.org/Member/wiki/FOdashboard)

**Comment by plehegar** [comment]
See [Proposed change to the Web Performance WG charter](https://lists.w3.org/Archives/Member/member-charters-review/2026Jul/0001.html)

**Comment by plehegar** [comment]
[Formal objection](https://lists.w3.org/Archives/Public/public-review-comments/2026Jul/0007.html)

**Comment by plehegar** [comment]
[Call for Participation: Web Performance Working Group Charter approved; Join the Web Performance Working Group](https://lists.w3.org/Archives/Public/public-new-work/2026Aug/0004.html)
